pkg/client/kubernetes/base/apply.go

```diff
@@ -128,8 +128,11 @@ func (c *Client) Apply(m []byte) error {
 				// We do not want to overwrite a ServiceAccount's `.secrets[]` list or `.imagePullSecrets[]`.
 				newObj.Object["secrets"] = oldObj.Object["secrets"]
 				newObj.Object["imagePullSecrets"] = oldObj.Object["imagePullSecrets"]
+			case "BackupInfrastructure":
```

Comment:
Apply.go while chart rendering we call put method on client which removes the finalizer since it it not in the template. So i have to handle it here. If you know the alternative please let me know.